cpp/src/rolling/rolling_detail.hpp

```diff
@@ -44,23 +44,26 @@ namespace detail
     constexpr bool timestamp_sum =
         std::is_same<AggOp, DeviceSum>::value &&
         cudf::is_timestamp<ColumnType>();
-
-
-    constexpr bool is_valid_rolling_agg = !std::is_same<ColumnType, cudf::string_view>::value and
-                                          !timestamp_mean and
-                                          !timestamp_sum and
-                                            ((op == experimental::aggregation::SUM) or
+    
+    constexpr bool is_operation_supported == (op == experimental::aggregation::SUM) or
                                              (op == experimental::aggregation::MIN) or
                                              (op == experimental::aggregation::MAX) or
                                              (op == experimental::aggregation::COUNT) or
-                                             (op == experimental::aggregation::MEAN)) and
-                                           (cudf::is_numeric<ColumnType>() or
-                                            comparable_countable_op);
+                                             (op == experimental::aggregation::MEAN)
+
+    constexpr bool is_valid_timestamp_agg = !timestamp_mean and !timestamp_sum and cudf::is_timestamp<ColumnType>();
+
+
+    constexpr bool is_valid_numeric_agg = cudf::is_numeric<ColumnType>() and
+                                          comparable_countable_op and 
+                                          is_operation_supported
+
+    constexpr bool is_valid_rolling_agg = !std::is_same<ColumnType, cudf::string_view>::value and
```

Comment:
Hold on, this seems to make all operations on strings columns invalid. Don't we support min, max and count on strings?

Reply:
There is one more check below which handles string only operations.
https://github.com/rapidsai/cudf/blob/c91db815be6968bacf2ecb7c01a059c31cacbe32/cpp/src/rolling/rolling_detail.hpp#L64

Reply:
Why not combine them?

Reply:
These two are being used to channel the code into two different launchers which handle the output differently. For string and for generic types.
https://github.com/rapidsai/cudf/blob/c91db815be6968bacf2ecb7c01a059c31cacbe32/cpp/src/rolling/rolling.cu#L319

https://github.com/rapidsai/cudf/blob/c91db815be6968bacf2ecb7c01a059c31cacbe32/cpp/src/rolling/rolling.cu#L292